src/components/OfflineSearch.ts

```diff
@@ -0,0 +1,173 @@
+import { Component, Props, createElement } from "react";
+import { findDOMNode } from "react-dom";
+
+import * as dijitRegistry from "dijit/registry";
+
+import { Alert } from "./Alert";
+import "../ui/OfflineSearch.css";
+
+type searchMethodOptions = "equals" | "lessThan" | "lessThanOrEquals" | "greaterThan" | "greaterThanOrEquals" | "contains";
+
+interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: string | null;
+        _setsize: number;
+        _setSize: number;
+        atEnd: () => boolean;
+        _pageSize: number;
+    };
+    _loadMore: () => void;
+    _onLoad: () => void;
+    _renderData: () => void;
+    update: () => void;
+}
+
+interface OfflineSearchProps extends Props<OfflineSearch> {
```

Comment:
Usage is depricated, so just remove. The previous purpose of this is already taken care of when u specify this as the props interface on the Component class. `Component<OfflineSearchProps, {}>`